src/GenericCyclotomics.jl

```diff
@@ -0,0 +1,462 @@
+using Oscar.Random: Random, SamplerTrivial, GLOBAL_RNG
+using Oscar.RandomExtensions: RandomExtensions, Make2, AbstractRNG
+
+import Oscar.AbstractAlgebra: parent_type, elem_type, base_ring, parent, is_domain_type,
+	is_exact_type, canonical_unit, isequal, divexact, zero!, mul!, add!, addeq!,
+	get_cached!, is_unit, characteristic, Ring, RingElem, expressify
+import Oscar: pretty, Indent, Dedent, terse, is_terse
+import Oscar.AbstractAlgebra: normal_form
+
+import Base: show, +, -, *, ^, ==, inv, isone, iszero, one, zero, rand, deepcopy_internal, hash
+
+const UPoly = Generic.UnivPoly{QQFieldElem, Generic.MPoly{QQFieldElem}}
```

Comment:
```suggestion
const UPoly = Generic.UnivPoly{QQFieldElem}
```

Reply:
When I do this I get the following when trying to construct a generic cyclotomic:
```
julia> S(1)
ERROR: MethodError: Cannot `convert` an object of type 
  AbstractAlgebra.Generic.FracFieldElem{AbstractAlgebra.Generic.UnivPoly{QQFieldElem, AbstractAlgebra.Generic.MPoly{QQFieldElem}}} to an object of type 
  AbstractAlgebra.Generic.FracFieldElem{AbstractAlgebra.Generic.UnivPoly{QQFieldElem, U} where U<:MPolyRingElem{QQFieldElem}}

Closest candidates are:
  convert(::Type{T}, ::T) where T
   @ Base Base.jl:84
  AbstractAlgebra.Generic.FracFieldElem{T}(::T, ::T) where T<:RingElem
   @ AbstractAlgebra ~/.julia/packages/AbstractAlgebra/88ogO/src/generic/GenericTypes.jl:915

Stacktrace:
 [1] setindex!(h::Dict{…}, v0::AbstractAlgebra.Generic.UnivPoly{…}, key0::AbstractAlgebra.Generic.FracFieldElem{…})
   @ Base ./dict.jl:367
 [2] Dict{AbstractAlgebra.Generic.FracFieldElem{…}, AbstractAlgebra.Generic.UnivPoly{…} where U<:MPolyRingElem{…}}(kv::Dict{AbstractAlgebra.Generic.FracFieldElem{…}, AbstractAlgebra.Generic.UnivPoly{…}})
   @ Base ./dict.jl:84
 [3] convert
   @ ./abstractdict.jl:568 [inlined]
 [4] GenericCharacterTables.GenericCyclo(f::Dict{AbstractAlgebra.Generic.FracFieldElem{…}, AbstractAlgebra.Generic.UnivPoly{…}}, parent::GenericCharacterTables.GenericCycloRing)
   @ GenericCharacterTables ~/Archiv/Computer/Programme/GenericCharacterTables.jl/src/GenericCyclotomics.jl:140
 [5] (::GenericCharacterTables.GenericCycloRing)(c::Int64)
   @ GenericCharacterTables ~/Archiv/Computer/Programme/GenericCharacterTables.jl/src/GenericCyclotomics.jl:341
 [6] top-level scope
   @ REPL[20]:1
Some type information was truncated. Use `show(err)` to see complete types.

```

Reply:
Then we probably have to do the version dance. Since `pkgversion` is not available on julia 1.6, we could do the following hack to detect the new AA version:
```
@static isdefined(AbstractAlgebra.Solve, :MatrixNormalFormTrait)
  const UPoly = Generic.UnivPoly{QQFieldElem, Generic.MPoly{QQFieldElem}}
else
  const UPoly = Generic.UnivPoly{QQFieldElem}
```

Reply:
> Since pkgversion is not available on julia 1.6, ...

I think we don't need 1.6 support. We're not even testing against 1.6.

> or something like that,...

Sigh... Ok, I hoped we could get around this.

Reply:
If 1.6 is not supported, the compat version should be adjusted.

Reply:
I think I've found a solution which should work with 1.6 as well. It's the alternative solution we had for AA 0.42 https://github.com/oscar-system/GenericCharacterTables.jl/pull/85.

Reply:
@fingolfin What do you think about julia 1.6 support and testing?

Reply:
I think I disabled testing on Julia 1.6 in CI at some point due to problems. I'll see what the current status is and if we can restore support, see PR #128.

Since OSCAR currently supports Julia >= 1.6, as 1.6 is the Julia LTS release. So it would be kinda nice if we could also support it. But only if we can do it with reasonable effort.

Note that Julia 1.10 is expected to become the next Julia LTS release pretty soon. And then OSCAR might follow suite after some time, which would also resolve this (but of course then we should definitely adjust `Project.toml` to require a newer Julia.

Reply:
Ok, so this PR seems to be compatible with 1.6.